shim.c

```diff
@@ -1483,11 +1483,16 @@ load_certs(EFI_HANDLE image_handle)
 	}
 
 	while (1) {
-		int old = buffersize;
+		UINTN old = buffersize;
 		efi_status = dir->Read(dir, &buffersize, buffer);
 		if (efi_status == EFI_BUFFER_TOO_SMALL) {
-			buffer = ReallocatePool(buffer, old, buffersize);
-			continue;
+			if (buffersize != old) {
```

Comment:
@dennis-tseng99 if you don't mind I'd like to keep this check just to cover more cases than buffersize set to 0.
Also I'm not sure what happens when calling UEFI AllocatePool() (used by ReallocatePool() internally) with size set to 0.

Reply:
Can this be made to be one commit?